### PR TITLE
improve compatibility for closure build

### PR DESCRIPTION
**Pull request by gengjiawen**
## Summary

java print some output to stderr, caused wrong error detect

```console
-- PLUGIN_ERROR (scripts/rollup/plugins/closure-plugin) --
Error: Picked up JAVA_TOOL_OPTIONS:  -Xmx3489m

    at /workspace/react/scripts/rollup/plugins/closure-plugin.js:16:16
    at ChildProcess.<anonymous> (/workspace/react/node_modules/google-closure-compiler/lib/node/closure-compiler.js:103:9)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1093:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
error Command failed with exit code 1.                                                                       
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## How did you test this change?

This will fail: https://gitpod.io/#https://github.com/facebook/react

This works: https://gitpod.io/#https://github.com/facebook/react/pull/25150

### CODE REVIEW

**Comment by gengjiawen** [comment]
@gaearon Can you take a look ? thx.

File: scripts/rollup/plugins/closure-plugin.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function compile(flags) {
   return new Promise((resolve, reject) => {
     const closureCompiler = new ClosureCompiler(flags);
     closureCompiler.run(function(exitCode, stdOut, stdErr) {
-      if (!stdErr) {
+      if (!stdErr || exitCode === 0) {
```
Comment (gaearon) [inline scripts/rollup/plugins/closure-plugin.js:13]: Should this be the only condition then? Why both?

Comment (gengjiawen) [reply scripts/rollup/plugins/closure-plugin.js]: yeap. I will make the change.

**Comment by gaearon** [comment]
Can you verify that both stdout and stderr always get printed? Both locally and on CI. If not please add logs.

**Comment by gengjiawen** [comment]
> Can you verify that both stdout and stderr always get printed? Both locally and on CI. If not please add logs.

I am not sure I get this, stdOut and stdErr is not printed in original logic. 

The stdout is build js content, not sure it's what wanted.

**Comment by gaearon** [comment]
Ah I see.

Could there be a case where `stdErr` is non-empty but `errorCode` is `0` so we miss some important information that previously would've gotten the build to reject?

**Comment by gengjiawen** [comment]
> Like, what if GCC prints some important warning but doesn't fail the build. Would we want to see it? Should we maybe fix the original problem that you're running into? Like presumably some env variable being set incorrectly.

Maybe this ?
```js
if (exitCode === 0) {
    if (!stdErr) {
        console.log(stdErr.toString())
    }
    resolve(stdOut);
}
```

Java and R is quite weirdo on this. They even print version in stderr. You can see https://sourcegraph.com/github.com/tabrindle/envinfo@main/-/blob/src/helpers/languages.js?L47

**Comment by xc1427** [comment]
I meet the same issue when using gitpod.io to build the source.

I think this issue may be due to some rare behavior of java in some Linux distro.

@gengjiawen could you please resolve the PR conflict?

**Comment by gengjiawen** [comment]
@xc1427 done. @gaearon Can you take another look ?

**Comment by xc1427** [comment]
> @xc1427 done. @gaearon Can you take another look ?

@gengjiawen 3q. Dan is no longer in facebook. Maybe ask someone else to review the PR ?

**Comment by gengjiawen** [comment]
> > @xc1427 done. @gaearon Can you take another look ?
> 
> @gengjiawen 3q. Dan is no longer in facebook. Maybe ask someone else to review the PR ?

I am in facebook. so have no idea when can review this.

**Pull request closed without merging**